Src/ILGPU/Context.cs

```diff
@@ -384,6 +385,59 @@ public DeviceCollection<TDevice> GetDevices<TDevice>()
                 : new DeviceCollection<TDevice>(new List<Device>());
         }
 
+        /// <summary>
+        /// Attempts to return the most optimal single device.
+        /// </summary>
+        /// <param name="preferCPU">Always returns CPU device 0.</param>
+        /// <returns>Selected device.</returns>
+        public Device GetPreferredDevice(bool preferCPU) =>
+            GetPreferredDevices(preferCPU, matchingDevicesOnly: false).First();
+
+        /// <summary>
+        /// Attempts to return the most optimal set of devices.
+        /// </summary>
+        /// <param name="preferCPU">Always returns first CPU device.</param>
+        /// <param name="matchingDevicesOnly">Only returns matching devices.</param>
+        /// <returns>Selected devices.</returns>
+        public IEnumerable<Device> GetPreferredDevices(
+            bool preferCPU, bool matchingDevicesOnly)
```

Comment:
nit: please break after the first parameter.

Reply:
Fixed, is a formatting guide linked anywhere for ILGPU?

Reply:
There is no explicit documentation. What I recommend, however, is to always look at the surrounding code (e.g. https://github.com/m4rs-mt/ILGPU/blob/c26dd77731ea2779c5c33ae9d91a74f85917978a/Src/ILGPU/Context.cs#L224-L228)